src/Microsoft.ApplicationInsights.AspNet/Extensibility/Implementation/Tracing/AspNetEventSource.cs

```diff
@@ -0,0 +1,126 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AspNetEventSource.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.ApplicationInsights.AspNet.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Diagnostics.Tracing;
+
+    /// <summary>
+    /// Event source for Application Insights ASP.NET 5 SDK.
+    /// </summary>
+    [EventSource(Name = "Microsoft-ApplicationInsights-AspNetCore")]
+    internal sealed class AspNetEventSource : EventSource
+    {
+        /// <summary>
+        /// The singleton instance of this event source.
+        /// Due to how EventSource initialization works this has to be a public field and not
+        /// a property otherwise the internal state of the event source will not be enabled.
+        /// </summary>
+        public static readonly AspNetEventSource Instance = new AspNetEventSource();
+
+        /// <summary>
+        /// Prevents a default instance of the AspNetEventSource class from being created.
+        /// </summary>
+        private AspNetEventSource() : base()
+        {
+            try
+            {
+                this.ApplicationName = Microsoft.Extensions.PlatformAbstractions.PlatformServices.Default.Application.ApplicationName;
+            }
+            catch (Exception exp)
+            {
+                this.ApplicationName = "Undefined " + exp.Message;
+            }
+        }
+
+        /// <summary>
+        /// Gets the application name for use in logging events.
+        /// </summary>
+        public string ApplicationName { [NonEvent] get; [NonEvent]private set; }
+
+        /// <summary>
+        /// Logs an event for the an exception in the TelemetryInitializerBase Initialize method.
+        /// </summary>
+        /// <param name="exception">The exception to write an event for.</param>
+        /// <param name="appDomainName">An ignored placeholder to make EventSource happy.</param>
+        [Event(1, Message = "{0}", Level = EventLevel.Error, Keywords = Keywords.Diagnostics)]
+        public void LogTelemetryInitializerBaseInitializeException(Exception exception, string appDomainName = "Incorrect")
```

Comment:
This override is very slow (https://msdn.microsoft.com/en-us/library/hh393360(v=vs.110).aspx). Do you think we can pass exception.Message, exception.StackTrace as 2 parameters here?